session.go

```diff
@@ -316,23 +356,34 @@ func (snd *sender) send(frame []byte) {
 	if err != nil {
 		snd.onSessionError(nil, err)
 	}
+
+	return
 }
 
-func (snd *sender) coalesceAdditionalFrames() {
+func (snd *sender) coalesceAdditionalFrames() bool {
 	// Coalesce enough to exceed coalesceThreshold
 	for snd.startOfData+snd.coalescedBytes+snd.cipherOverhead < coalesceThreshold {
 		select {
-		case frame := <-snd.out:
+		case frame, more := <-snd.out:
```

Comment:
It seems like `case frame, open := <-snd.out` here would be more idiomatic and clearer. Like we're just using the closed state of the channel here to determine if there's more data, which `more` seems to cloud

Reply:
Fixed.